plugin/storage/es/factory_test.go

```diff
@@ -259,6 +259,7 @@ func TestArchiveEnabled(t *testing.T) {
 	r, err := f.CreateArchiveSpanReader()
 	require.NoError(t, err)
 	assert.NotNil(t, r)
+	defer f.Close()
```

Comment:
Calling `defer` as the last statement in the function is completely pointless, you could just as well call the function directly. Instead, you want to call defer right after `f.Initialize() + require.NoError`, because it's then the resources might have been acquired by the factory (but check for error first)

Reply:
okay